File: python/bubble_sort.py
from base_algorithm import BaseAlgorithm
from isorting_algorithm import ISortingAlgorithm

class BubbleSort(BaseAlgorithm, ISortingAlgorithm):
    def __init__(self):
        super().__init__()
        self.name = "Ordenamiento Burbuja"
        self.minimal_complexity = "O(n^2)"
        self.average_complexity = "O(n^2)"
        self.maximal_complexity = "O(n^2)"

    def sort(self, array):
        n = len(array)
        for i in range(n - 1):
            swapped = False
            for j in range(n - i - 1):
                if array[j] > array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]
                    swapped = True
            if not swapped:
                break

    def get_elemental_operations(self, n):
        return n * n // 2

File: src/sorting/BubbleSort.java
package sorting;

import base.BaseAlgorithm;
import interfaces.ISortingAlgorithm;

public class BubbleSort extends BaseAlgorithm implements ISortingAlgorithm {

    public BubbleSort(){
        this.name = "Ordenamiento Burbuja";
        this.minimalComplexity = "O(n^2)";
        this.averageComplexity = "O(n^2)";
        this.maximalCompleity = "O(n^2)";

    }

    @Override
    public void sort(int[] array) {
        int n = array.length;
        boolean swapped;

        for (int i = 0; i < n -1; i++) {
            swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int temp = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = temp;
                    swapped = true;
                }
            }
            if (!swapped) break;
        }

    }

    @Override
    public Long getElementalOperations(int n) {
        return (long) n * n / 2;
    }
}
